app/models/category.rb

```diff
@@ -21,4 +21,9 @@ class Category < ApplicationRecord
   def full_name
     "#{ parent&.name&.upcase + ": " if parent}#{parent.present? ? name : name.upcase }"
   end
+
+  def lineage
+    own = [name]
+    parent ? parent.lineage + own : own
+  end
```

Comment:
I'm not satisfied with this method name. I was looking for something that meant "ancestors + self"... Open to ideas!

Reply:
oooo. i like it!